datafusion/optimizer/src/projection_push_down.rs

```diff
@@ -527,7 +527,9 @@ fn optimize_plan(
 }
 
 fn projection_equal(p: &Projection, p2: &Projection) -> bool {
-    p.expr.len() == p2.expr.len() && p.expr.iter().zip(&p2.expr).all(|(l, r)| l == r)
+    p.expr.len() == p2.expr.len()
+        && p.alias == p2.alias
```

Comment:
FYI @andygrove a small change because otherwise it also removes projections with (different) alias.